src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResource.java

```diff
@@ -547,6 +552,33 @@ public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithAuxiliaryRe
         return ResponseEntity.ok(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises/project-key/:projectKey : Queries a programming exercise by its project key.
+     *
+     * @param projectKey the project key of the programming exercise
+     *
+     * @return the ProgrammingExercise with this project key in an ResponseEntity or 404 Not Found if no exercise exists. It includes the students participation, submissions,
+     *         results and feedbacks.
+     */
+    @GetMapping("programming-exercises/project-key/{projectKey}")
+    @EnforceAtLeastStudent
+    @AllowedTools(ToolTokenType.SCORPIO)
+    public ResponseEntity<ProjectKeyProgrammingExerciseDTO> getProgrammingExerciseByProjectKey(@PathVariable String projectKey) {
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+
+        final ProgrammingExercise exercise = programmingExerciseRepository.findAllByProjectKey(projectKey).stream().findAny()
+                .orElseThrow(() -> new EntityNotFoundException("ProgrammingExercise", projectKey));
+
+        if (!authCheckService.isAllowedToSeeCourseExercise(exercise, user)) {
+            throw new AccessForbiddenException("You are not allowed to access this exercise");
+        }
+
+        var participations = studentParticipationRepository.findAllByExerciseIdAndStudentIdWithEagerLatestSubmissionLatestResultFeedbacksTestCases(exercise.getId(), user.getId());
```

Comment:
You still need to make sure that the student is allowed to see the latest result. There are cases when this should not be possible, e.g. in exams when the results are not published yet or when there are manual results in course exercise and the assessment due date is not over yet. It probably makes more sense to reuse some existing repository query instead of creating a new one